elements/nuxeo-dropzone/nuxeo-dropzone.js

```diff
@@ -491,4 +529,22 @@ Polymer({
   _showAbort(uploading) {
     return uploading && this.hasAbort();
   },
+
+  _getValidity() {
+    if (this.uploading) {
+      this._errorMessage = this.i18n('dropzone.invalid.uploading');
+      return false;
+    }
+    if (!this.required) {
+      return true;
+    }
+    return this.files && this.files.length > 0;
+  },
+
+  _handleError(e) {
+    if (e.detail && e.detail.error && e.detail.error.response && e.detail.error.response.status) {
```

Comment:
I would expect to have something like:
```suggestion
    if (this.invalid && e.detail && e.detail.error && e.detail.error.response && e.detail.error.response.status) {
```
In the sense that there shouldn't be anything preventing me from submitting a form **if** the upload failed by whatever reason, and this is not a required field.

Is this a shortcut to display an error message, which would probably be similar to what we need to achieve under NXP-27982?

Reply:
>Is this a shortcut to display an error message, which would probably be similar to what we need to achieve under NXP-27982?

Not really, here an error may occur during batch upload is hapenning. Typically:
 - using s3 direct upload, it is s3 bucket that would return a potential error
 - deploying the web ui in its own nginx with implicit `body-size` limit (see https://github.com/nuxeo/nuxeo-web-ui/commit/3c31848a8a2daa58afa83211cca802860b2993ea)

All these use cases may return errors that cannot be handled like the validation report after a POST or a PUT on the document itself.

If such a scenario occurs, we need to notify the user the blob wasn't successfully uploaded and clicking `Save` or `Create` won't attach the blob to the document. So setting the dropzone invalid is a way to make sure:
 1- submit does not happen
 2- the form focuses on the dropzone element
 3- the user sees the upload failed and eventually why
 4- the user is incited to retry (e.g. with a smaller file if size exceeded), clear and submit without attachment, etc.

Reply:
Sorry, I didn't mean using form hooks. What I meant was that we are lacking a system to display error messages on fields, which IIRC, was what we initially wanted to do on [NXP-27982](https://jira.nuxeo.com/browse/NXP-27982) but couldn't, because we were also trying to do it leveraging validity. The validity is used to check for value format using constraints, but in the case of an error, the value remains as is.

Anw, I can live with this as a temporary shortcut, although this does not seem to be necessary to solve [NXP-27803](https://jira.nuxeo.com/browse/NXP-27803). But if indeed we'd like to go in the direction of having a way to display error messages on fields not related to validation, then I'd wait for it or at least add a comment stating that it should be cleaned up once that feature is implemented.